README.md

```diff
@@ -14,148 +14,498 @@ referred to as AuthZ) and is what we'll discuss in this section.
+slightly in the `SecurityConfiguration` class to add authorization!
+
+We can use the "request matching" functionality we used for AuthN, but instead
+of coupling it with `authenticated()` or `permitAll()` like we did in the
+previous section, we're going to couple it with the `hasAuthority()` method and
+pass it the name of the "authority" that we want to restrict access to.
+
+If we remember, when inserting new data into the database, we gave the user
+`mary` the `read` authority but the `admin` user an `admin` authority. We only
+want the `admin` to be able to access our `/status` endpoint, but we'll still
+leave the `/hello` endpoint public so anyone can access the greeting message.
+
+Modify the `filterChain()` method as such:
+
+```java
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity httpSecurity) throws Exception {
+        httpSecurity.httpBasic().and().authorizeRequests().antMatchers("/status").hasAuthority("admin");
+        httpSecurity.httpBasic().and().authorizeRequests().antMatchers("/hello").permitAll();
+        httpSecurity.httpBasic().and().authorizeRequests().anyRequest().authenticated();
+        return httpSecurity.build();
     }
 ```
 
-Restart your application now and observe that you will only be able to access
-the `/admin` URL with the "admin" user.
+So what did we do here? We added the line
+`httpSecurity.httpBasic().and().authorizeRequests().antMatchers("/status").hasAuthority("admin");`
+to say that in order to access this endpoint, the authenticated user _must_ have
+an admin authority. If the user is not authenticated and/or does not have the
+admin authority attached to it, then we will not be able to reach it.
 
 As with authentication rules, authorization rules can also be chained and should
 be ordered from most specific to least specific.
 
-In our example, we could require the "admin" authority for both the `/status`
-and `/hello` URLs with the following `configure()` method:
+Let's test this out! Open up Postman and start up the application.
+
+We'll start off by choosing the "No Auth" type in the "Authorization" tab in
+Postman and sending the "http://localhost:8080/hello?name=mary" GET request URL
+to ensure that it still works without having to be authenticated:
+
+![postmnan-no-auth](https://curriculum-content.s3.amazonaws.com/spring-mod-2/authentication/postman-no-authentication.png)
+
+Now let's try to hit the "http://localhost:8080/status" endpoint without any
+authentication.
+
+![postman-401-unauthorized](https://curriculum-content.s3.amazonaws.com/spring-mod-2/authentication/postman-unauthorized.png)
+
+Everything is working great still! We'll turn on the "Basic Auth" now and enter
+the credentials for `mary` while hitting the "/status" endpoint. We should be
+forbidden from seeing the endpoint with this user:
+
+![postman-403-forbidden](https://curriculum-content.s3.amazonaws.com/spring-mod-2/authorization/postman-forbidden.png)
```

Comment:
Was the meaning of 401 and 403 explained in a previous lesson?

Reply:
Status codes are briefly mentioned but I will add a note explaining the differences in this lesson.